### PR TITLE
Mirror api fix

### PR DESCRIPTION
**Pull request by MarkSymsCtx**

### CODE REVIEW

**Comment by MarkSymsCtx** [comment]
as discussed with @edwintorok it turns out that these bits were unusable from the generated python due to type confusion. As such it is highly unlikely that anyone has used them so a breaking change is in this case acceptable

**Comment by lindig** [comment]
I am surprised that these changes don't require corresponding changes of existing client code.

**Review by edwintorok — APPROVED** [review]
'progress = Some 1.0'  is probably the same as 'complete = true', but it is probably better to be explicit here, otherwise some floating point rounding error might lead to us to believe that an operation is complete before it really is.

File: ocaml/xapi-storage/generator/lib/data.ml [hunk 1]
```diff
@@ -239,14 +239,18 @@ end
...
+
   (** The primary key for referring to a long-running operation. *)
   type operation =
-    | Copy of uri * uri
-        (** Copy (src,dst) represents an on-going copy operation
-        from the [src] URI to the [dst] URI. *)
-    | Mirror of uri * uri
-        (** Mirror (src,dst) represents an on-going mirror
-        operation from the [src] URI to the [dst] URI. *)
+    | CopyV1 of copy_operation_v1
```
Comment (Vincent-lau) [inline ocaml/xapi-storage/generator/lib/data.ml:248]: why is there a v1?

File: ocaml/xapi-storage/generator/lib/data.ml [hunk 2]
```diff
@@ -239,14 +239,18 @@ end
...
+        (** CopyV1 (key) represents an on-going copy operation
+        with the unique [key]. *)
```
Comment (Vincent-lau) [inline ocaml/xapi-storage/generator/lib/data.ml:250]: what is this key representing?

File: ocaml/xapi-storage/generator/lib/data.ml (hunk 1 above)
Comment (MarkSymsCtx) [reply ocaml/xapi-storage/generator/lib/data.ml:248]: at @edwintorok 's recommendation to allow for future expansion

File: ocaml/xapi-storage/generator/lib/data.ml (hunk 2 above)
Comment (MarkSymsCtx) [reply ocaml/xapi-storage/generator/lib/data.ml:250]: it's an abstract key that can be passed back to `stat` to obtain information about the running operation.

**Comment by MarkSymsCtx** [comment]
> I am surprised that these changes don't require corresponding changes of existing client code.

It was when we tried to wire up the Python side of this that we discovered that it couldn't actually be used as specified as the types in the ocaml can't be represented in JSON and so results from an operation can't be round-tripped and used as input for a subsequent API call.

File: ocaml/xapi-storage/generator/lib/data.ml (hunk 2 above)
Comment (Vincent-lau) [reply ocaml/xapi-storage/generator/lib/data.ml:250]: so something like a handle/token

**Review by Vincent-lau — APPROVED** [review]

**Review by robhoes — APPROVED** [review]

**Pull request merged**